app/dv/src/main/java/io/syndesis/dv/lsp/parser/DdlAnalyzerConstants.java

```diff
@@ -0,0 +1,537 @@
+        CREATE_VIRTUAL_VIEW_TYPE
+    }
+
+    enum CONTEXT {
+        NONE_FOUND,
+        PREFIX,
+        TABLE_BODY,
+        TABLE_OPTIONS,
+        TABLE_ELEMENT,
+        TABLE_ELEMENT_OPTIONS,
+        TABLE_ELEMENT_OPTION_SEARCHABLE,
+        QUERY_EXPRESSION,
+//        AS_TOKEN,
+        SELECT_CLAUSE,
+        FROM_CLAUSE,
+        WHERE_CLAUSE
+    }
+
+    int[] DATATYPES = {
+        STRING,
+        VARBINARY,
+        VARCHAR,
+        BOOLEAN,
+        BYTE,
+        TINYINT,
+        SHORT,
+        SMALLINT,
+        CHAR,
+        INTEGER,
+        LONG,
+        BIGINT,
+        BIGINTEGER,
+        FLOAT,
+        REAL,
+        DOUBLE,
+        BIGDECIMAL,
+        DECIMAL,
+        DATE,
+        TIME,
+        TIMESTAMP,
+        BLOB,
+        CLOB,
+        XML,
+        JSON,
+        GEOMETRY,
+        GEOGRAPHY,
+        OBJECT
+    };
+
+    String[] DATATYPE_LIST = {
+        getLabel(STRING, false),
+        getLabel(VARBINARY, false),
+        getLabel(VARCHAR, false),
+        getLabel(BOOLEAN, false),
+        getLabel(BYTE, false),
+        getLabel(TINYINT, false),
+        getLabel(SHORT, false),
+        getLabel(SMALLINT, false),
+        getLabel(CHAR, false),
+        getLabel(INTEGER, false),
+        getLabel(LONG, false),
+        getLabel(BIGINT, false),
+        getLabel(BIGINTEGER, false),
+        getLabel(FLOAT, false),
+        getLabel(REAL, false),
+        getLabel(DOUBLE, false),
+        getLabel(BIGDECIMAL, false),
+        getLabel(DECIMAL, false),
+        getLabel(DATE, false),
+        getLabel(TIME, false),
+        getLabel(TIMESTAMP, false),
+        getLabel(BLOB, false),
+        getLabel(CLOB, false),
+        getLabel(XML, false),
+        getLabel(JSON, false),
+        getLabel(GEOMETRY, false),
+        getLabel(GEOGRAPHY, false),
+        getLabel(OBJECT, false)
+    };
+
+    /*
+     * Array of tokens that match the start of a CREATE TABLE statement
+     */
+    int[] CREATE_TABLE_STATEMENT = { CREATE, TABLE };
+
+    /*
+     * Array of tokens that match the start of a CREATE FOREIGNT TABLE statement
+     */
+    int[] CREATE_FOREIGN_TABLE_STATEMENT = { CREATE, FOREIGN, TABLE };
+
+    /*
+     * Array of tokens that match the start of a CREATE FOREIGN TEMPORARY TABLE
+     * statement
+     */
+    int[] CREATE_FOREIGN_TEMPORARY_TABLE_STATEMENT = { CREATE, FOREIGN, TEMPORARY, TABLE };
+
+    /*
+     * Array of tokens that match the start of a CREATE TEMPORARY TABLE
+     * statement
+     */
+    int[] CREATE_GLOBAL_TEMPORARY_TABLE_STATEMENT = { CREATE, GLOBAL, TEMPORARY, TABLE };
+
+    /*
+     * Array of tokens that match the start of a CREATE VIEW
+     * statement
+     */
+    int[] CREATE_VIEW_STATEMENT = { CREATE, VIEW };
+
+    /*
+     * Array of tokens that match the start of a CREATE VIRTUAL VIEW
+     * statement
+     */
+    int[] CREATE_VIRTUAL_VIEW_STATEMENT = { CREATE, VIRTUAL, VIEW };
+
+    int[] NON_DATATYPE_KEYWORDS = {
```

Comment:
Can we remove this list and just drive it off of SQLContants?  We add keywords more often than datatypes, so if we we need the datatypes hardcoded for now that's fine.  Also if there is ever consideration for domain types - which can use non-reserved keyword names - then this may need some refinement.

Reply:
Since some of the datatypes required SNIPPET definition, it made sense to define them like other completion items (*.json) and lazily load them when needed. If you have an idea on how to generate the completion items from SQLConstants, then log a jira and we'll replace the array.

Note that the "int" value of all of the SQLConstants, defines the token KIND which is used extensively during the completion item collection and diagnostics process via the TokenContext.

Reply:
Please see https://github.com/blafond/syndesis/pull/6 - if there are specific exclusions you want, that could be hard-coded.

Separating the keywords to datatype and non-datatype may not be what you need to do - you will still need datatypes in the completion list for cast/conversion functions - cast, convert, xmlcast, etc.